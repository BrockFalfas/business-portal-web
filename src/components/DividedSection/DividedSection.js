import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import './DividedSection.scss';

const Cell = ({ children }) => <div className="DividedSection-cell">{children}</div>;

export class DividedSection extends React.Component {
  static Cell = Cell;

  static propTypes = {
    children: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
    orientation: PropTypes.oneOf(['horizontal', 'vertical']),
  };

  static defaultProps = {
    orientation: 'horizontal',
  };

  render() {
    const { children, orientation } = this.props;

    return (
      <div className={classnames('DividedSection', { [`DividedSection--${orientation}`]: true })}>
        {children}
      </div>
    );
  }
}

export default DividedSection;
