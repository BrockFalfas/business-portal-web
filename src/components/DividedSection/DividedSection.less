@import "../../styles/variables.less";

.DividedSection {
  display: flex;
  justify-content: space-between;
  align-items: stretch;

  &--horizontal {
    flex-direction: row;

    .DividedSection-cell + .DividedSection-cell {
      border-left: 1px solid @color-grey;
    }
  }

  &--vertical {
    flex-direction: column;

    .DividedSection-cell + .DividedSection-cell {
      border-top: 1px solid @color-grey;
    }
  }

  &-cell {
    flex: 1;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
  }
}
