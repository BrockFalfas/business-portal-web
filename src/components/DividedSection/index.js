import DividedSection from './DividedSection';

export default DividedSection;
