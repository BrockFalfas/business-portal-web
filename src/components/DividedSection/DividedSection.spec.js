import React from 'react';
import { shallow } from 'enzyme';

import DividedSection from './index';

const initDividedSection = overrides => {
  const wrapper = shallow(<DividedSection {...overrides} />);
  return { wrapper };
};

describe('component: <DividedSection/>', () => {
  describe('DividedSection', () => {
    it('should render without crashing', () => {
      const { wrapper } = initDividedSection();
      expect(wrapper).toMatchSnapshot();
    });

    it('should render classname based on orientation', () => {
      const { wrapper } = initDividedSection({ orientation: 'vertical' });
      expect(wrapper.prop('className')).toContain('DividedSection--vertical');

      wrapper.setProps({ orientation: 'horizontal' });
      expect(wrapper.prop('className')).toContain('DividedSection--horizontal');
    });
  });
});
