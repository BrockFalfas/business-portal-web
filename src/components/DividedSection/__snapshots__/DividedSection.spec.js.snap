// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: <DividedSection/> DividedSection should render without crashing 1`] = `
<div
  className="DividedSection DividedSection--horizontal"
/>
`;
